taiga2/controllers/endpoint.py

```diff
@@ -618,8 +687,29 @@ def create_new_dataset_version(datasetVersionMetadata):
     new_description = datasetVersionMetadata["newDescription"]
     changes_description = datasetVersionMetadata.get("changesDescription", None)
 
-    new_dataset_version = models_controller.create_new_dataset_version_from_session(
-        session_id, dataset_id, new_description, changes_description
+    # Older versions of taigapy don't include add_existing_files in the params. Instead,
+    # add_existing_files is handled within the taigapy logic. So it's safe to default this
+    # to False and avoid repeating this logic.
+    add_existing_files = datasetVersionMetadata.get("addExistingFiles", False)
+
+    # Only used if add_existing_files is True to support grabbing the existing files
+    dataset_version = datasetVersionMetadata.get("datasetVersion", None)
+    current_user = models_controller.get_current_session_user()
+
+    # The next 2 lines serve the purpose of providing a datafile names list that we
+    # later use for comparison with previous version datafiles (if add_existing_files is true).
+    added_files = models_controller.get_upload_session_files_from_session(session_id)
```

Comment:
It looks like added_files and datafile_names are only necessary if `add_existing_files == True`

If that's the case, is there a reason they're set outside of `create_new_dataset_version_from_session`? It looks like it'd be more natural to scope them to inside that function given that they're not used outside of that function.